### PR TITLE
[TH2-4985] Update to `th2-grpc-common` version `4.2.0`

### PR DESCRIPTION
**Pull request by Nikita-Smirnov-Exactpro**
+ Update to `th2-bom` version `4.4.0`
+ Update to `th2-grpc-service-genrator` version `3.4.0`

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -67,20 +55,6 @@ dependencies {
...
-    main {
-        proto {
-            srcDirs 'src/main/proto'
-        }
-        java {
-            srcDirs 'src/gen/main/java', 'src/gen/main/grpc', 'src/gen/main/services/java/src'
-        }
-        resources {
-            srcDirs 'src/gen/main/services/java/resources'
-        }
```
Comment (OptimumCode) [inline build.gradle:80]: This must be left and destination corrected to the place where the updated protobuf generator places the generated source

File: build.gradle [hunk 2]
```diff
@@ -67,20 +55,6 @@ dependencies {
...
-    }
-}
-
 protobuf {
```
Comment (OptimumCode) [inline build.gradle:58]: We don't need this
`generateProtoTasks.generatedFilesBaseDir = genDir`
the updated version of protobuf plugin puts the generated source to `build/generated` and add them as source set.
If we leave this property here it causes files duplication and you had to fix that (line 111)

**Review by OptimumCode — CHANGES_REQUESTED** [review]

File: build.gradle [hunk 3]
```diff
@@ -55,6 +54,14 @@ dependencies {
     api "com.exactpro.th2:grpc-service-generator:${serviceGeneratorVersion}"
 }
 
+sourceSets {
+    main {
+        resources {
+            srcDirs 'src/gen/main/services/java/resources'
```
Comment (OptimumCode) [inline build.gradle:60]: Have you checked the generation? I believe they are located in a different directory
See the example here
https://github.com/th2-net/th2-grpc-sim/blob/dev-version-5/build.gradle#L42

**Review by OptimumCode — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 3 above)
Comment (Nikita-Smirnov-Exactpro) [reply build.gradle]: My bad, thank you for recheck.

**Review by OptimumCode — APPROVED** [review]

**Review by OptimumCode — APPROVED** [review]

File: build.gradle [hunk 4]
```diff
@@ -40,8 +40,8 @@ repositories {
 }
 
 dependencies {
-    api platform('com.exactpro.th2:bom:4.4.0-update-libs-5345334944-c71fdc9-SNAPSHOT')
-    api 'com.exactpro.th2:grpc-common:4.3.0-TH2-4887-vulnerabilities-5374522292-1f54a29-SNAPSHOT'
+    api platform('com.exactpro.th2:bom:4.4.0')
+    api 'com.exactpro.th2:grpc-common:4.3.0-TH2-4887-vulnerabilities-+'
```
Comment (OptimumCode) [inline build.gradle:44]: Do we really want to go with version wildcard?

Comment (Nikita-Smirnov-Exactpro) [reply build.gradle:44]: I very tired to update the versions in ten+ repositories, also this feature must be released sooner or later.

Comment (OptimumCode) [reply build.gradle:44]: Okay...we have chosen the mulirepo way by ourselves) And here are the consequences...

**Review by OptimumCode — APPROVED** [review]

**Pull request merged**